src/components/Article.js

```diff
@@ -1,22 +1,87 @@
 import styles from "./article.module.scss";
+import { Link, useSearchParams } from "react-router-dom";
+import starImage from "images/star-svgrepo-com.svg";
+import inactiveStarImage from "../images/star-inactive.com.svg";
+import { useState } from "react";
 
-export default function Articles({ article }) {
+export default function Articles({ article, articleIndex, pageNumber }) {
   const date = article.publishedAt.slice(0, 10);
+  const [searchParams] = useSearchParams();
+  const [buttonImage, setButtonImage] = useState(
+    JSON.parse(localStorage.getItem("favorites")) &&
+      JSON.parse(localStorage.getItem("favorites")).findIndex(
+        (element) => element.title === article.title
+      ) !== -1
+      ? starImage
+      : inactiveStarImage
+  );
 
   return (
     <div className={styles["article"]}>
-      <h2 className={styles["article__title"]}>{article.title}</h2>
-      <div className={styles["article__content"]}>
+      <button
+        className={styles["article__favorite-button"]}
+        onClick={() => {
+          if (!localStorage.getItem("favorites")) {
+            localStorage.setItem("favorites", "[]");
+          }
+
+          const favoritesStorage = JSON.parse(
+            localStorage.getItem("favorites")
+          );
+
+          if (
+            favoritesStorage.findIndex(
```

Comment:
You can create a variable for the found index and reuse it since you need index in 2 places